src/movie/movie.service.ts

```diff
@@ -0,0 +1,185 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository, In } from 'typeorm';
+import { CreateMovieDto } from './dto/create-movie.dto';
+import { UpdateMovieDto } from './dto/update-movie.dto';
+import { Movie } from '../entity/movie.entity';
+import { Director } from '../entity/director.entity';
+import { Actor } from '../entity/actor.entity';
+import { Festival } from '../entity/festival.entity';
+import { Character } from '../entity/character.entity';
+import { Genre } from '../entity/genre.entity';
+
+@Injectable()
+export class MovieService {
+  constructor(
+    @InjectRepository(Movie)
+    private readonly movieRepository: Repository<Movie>,
+    @InjectRepository(Director)
+    private readonly directorRepository: Repository<Director>,
+    @InjectRepository(Actor)
+    private readonly actorRepository: Repository<Actor>,
+    @InjectRepository(Festival)
+    private readonly festivalRepository: Repository<Festival>,
+    @InjectRepository(Character)
+    private readonly characterRepository: Repository<Character>,
+    @InjectRepository(Genre)
+    private readonly genreRepository: Repository<Genre>,
+  ) {}
+
+  async create(createMovieDto: CreateMovieDto): Promise<Movie> {
+    const { directorId, actorIds, festivalIds, genreIds, ...movieData } = createMovieDto;
+
+    const director = directorId ? await this.directorRepository.findOne({ where: { id: +directorId } }) : null;
+    const actors = actorIds && actorIds.length ? await this.actorRepository.find({ where: { id: In(actorIds) } }) : [];
+    const festivals = festivalIds && festivalIds.length ? await this.festivalRepository.find({ where: { id: In(festivalIds) } }) : [];
+    const genres = genreIds && genreIds.length ? await this.genreRepository.find({ where: { id: In(genreIds) } }) : [];
```

Comment:
Why do you need to get these entities from the DB before insertion. You can create the movie entity using the IDs.